### ISSUE TITLE
How to do array of enums?

### ISSUE THREAD

**Issue by IhsanMujdeci** [marcj/marshal.ts#41]
Hey I can't quite figure out the syntax, could I please get some help?

**Comment by IhsanMujdeci** [comment]
This doesn't work to cover the case of an array of enum
```
 @f.enum(EmploymentTypes)
    private employmentTypes: EmploymentTypes[] = [];
```

**Comment by marcj** [comment]
Try `@f.enum(Foo).asArray()`

**Comment by IhsanMujdeci** [comment]
Thanks mate worked, also thanks for this really cool library. Love the use of metadata. I don't know why this doesn't have 5k stars.
Could you please document this feature.

**Comment by IhsanMujdeci** [comment]
working code
```
@f.enum(EmploymentTypes).asArray()
private employmentTypes: EmploymentTypes[] = [];
```

**Comment by IhsanMujdeci** [comment]
Actually it doesn't work when i use `validatedPlainToClass` the field gets set to a string value and not an array of strings/

**Comment by IhsanMujdeci** [comment]
Also the validation complains when I try and put more that 1 enum value in the array. which isn't desired behaviour

**Comment by IhsanMujdeci** [comment]
Made a custom validator
```
function enumArray(theEnum) {
    return class implements PropertyValidator {
        validate<T>(value: any = [], propertySchema: PropertyCompilerSchema): PropertyValidatorError {
            if(!Array.isArray(value)){
                return new PropertyValidatorError('Value not an array', 'Was expecting array of enums to be array');
            }
            const enumValues: string[] = Object.values(theEnum);
            for(const v of value){
                if(!enumValues.find(x=> x===v)){
                    return new PropertyValidatorError('Not in enum', `Expected to see [${enumValues.join(', ')}] in array, but found ${v}`);
                }
            }
        };
    }
}
```

Class
```
@f.any().validator(enumArray(EmploymentTypes))
private employmentTypes: EmploymentArrays = [];
```

**Comment by marcj** [comment]
Oh, indeed, array enums was never tested or implemented. That's fixed in 2.1.6. You can now use `@f.enum(Foo).asArray()`

**Comment by IhsanMujdeci** [comment]
Tested it, seems to be working as expected. Thanks for letting me know about the fix.
